File: LeetCode/Python Solutions/Single Number II/single.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        """
        Calculate the single number in the given list.

        Parameters:
            nums (List[int]): A list of integers.

        Returns:
            int: The single number in the list.
        """
        once = twice = 0
        for num in nums:
            once = (once ^ num) & ~twice
            twice = (twice ^ num) & ~once
        return once

File: LeetCode/Java Solutions/Single Number II/single.java
class Solution {
    /**
     * Finds the single number in the given array of integers.
     *
     * @param  nums  an array of integers
     * @return       the single number in the array
     */
    public int singleNumber(int[] nums) {
        int once = 0, twice = 0;
        for (int num : nums) {
            once = (once ^ num) & ~twice;
            twice = (twice ^ num) & ~once;
        }
        return once;
    }
}

